ethereum/datastructures/src/testFixtures/java/tech/pegasys/teku/datastructures/util/DataStructureUtil.java

```diff
@@ -380,6 +380,13 @@ public DepositWithIndex randomDepositWithIndex() {
         randomUnsignedLong().mod(UnsignedLong.valueOf(Constants.DEPOSIT_CONTRACT_TREE_DEPTH)));
   }
 
+  public DepositWithIndex randomDepositWithIndex(long depositIndex) {
```

Comment:
it would still be helpful to be able to set the index to know what is purged, but i can update the function i'm not using?

Reply:
Yeah I'd keep the new function as-is, just instead of having duplicate code creating `DepositWithIndex` we'd have:
```
public DepositWithIndex randomDepositWithIndex() {
    return randomDepositWithIndex(randomLong());
}
```
followed by your new version that takes the depositIndex as a parameter.

Reply:
that makes perfect sense. done.